src/Share/NamespaceDiffs.hs

```diff
@@ -210,6 +277,16 @@ diffAtPathTermDiffs_ f (DiffAtPath {termDiffsAtPath, typeDiffsAtPath}) =
     & (Set.traverse . definitionDiffDiffs_) %%~ f
     <&> \termDiffsAtPath -> DiffAtPath {typeDiffsAtPath, termDiffsAtPath}
 
+witherDiffAtPathTermDiffs ::
+  (Monad f, Ord termDiff', Ord referent, Ord renderedTerm) =>
+  (termDiff -> f (Maybe termDiff')) ->
+  DiffAtPath referent reference renderedTerm renderedType termDiff typeDiff ->
+  f (DiffAtPath referent reference renderedTerm renderedType termDiff' typeDiff)
+witherDiffAtPathTermDiffs f DiffAtPath {termDiffsAtPath, typeDiffsAtPath} =
+  DiffAtPath
+    <$> Set.forMaybe termDiffsAtPath (runMaybeT . (definitionDiffDiffs_ (MaybeT . f)))
```

Comment:
Not positive, and not really a big deal, but you may be able to downgrade the `Monad` constraint to `Applicative` if you use `Compose` rather than `MaybeT` here?